utility_analysis_new/dp_engine.py

```diff
@@ -21,6 +25,70 @@
 import utility_analysis_new.combiners as utility_analysis_combiners
 
 
+@dataclasses.dataclass
+class MultiParameterConfiguration:
+    """Specifies parameters for multi-parameter Utility Analysis.
+
+    All MultiParameterConfiguration attributes corresponds to attributes in
+    pipeline_dp.AggregateParams.
+
+    UtilityAnalysisEngine can perform utility analysis for multiple sets of
+    parameters simultaneously. API for this is the following:
+    1. Specify blue-print AggregateParams instance.
+    2. Set MultiParameterConfiguration attributes which should have different
+    values.
+    3. Pass created objects to UtilityAnalysisEngine.aggregate().
+
+    Example:
+        max_partitions_contributed = [1, 2]
+        max_contributions_per_partition = [10, 11]
+
+        Then the utility analysis will be performed for
+          AggregateParams(max_partitions_contributed=1, max_contributions_per_partition=10)
+          AggregateParams(max_partitions_contributed=2, max_contributions_per_partition=11)
+    """
+    max_partitions_contributed: Sequence[int] = None
+    max_contributions_per_partition: Sequence[int] = None
+    min_sum_per_partition: Sequence[float] = None
+    max_sum_per_partition: Sequence[float] = None
+
+    def __post_init__(self):
+        field_values = dataclasses.asdict(self)
```

Comment:
nit: using the same notation ("attribute"-s or "field"-s) would probably make it slightly easier to read.